docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst

```diff
@@ -0,0 +1,303 @@
+your original configuration.
+
+.. note::
+
+   For more information about ``DeepSpeedTrial`` see :ref:`deepspeed-api`
+
+To facilitate this process, you must add a ``deepspeed_config`` field under the ``hyperparameters``
+section of your experiment. This field specifies the relative path to the DS ``json`` configuration
+file (written following the `DS documentation here <https://www.deepspeed.ai/docs/config-json/>`_)
+and is how ``dsat`` is informed of your default settings. For example, if your default DeepSpeed
+configuration is stored in ``ds_config.json`` at the top-level of your model directory, your
+``hyperparameters`` section should include:
+
+.. code:: yaml
+
+   hyperparameters:
+     deepspeed_config: ds_config.json
+
+Once this configuration is in place, accessing the appropriate settings dictionary for each trial is
+straightforward. You can use the :func:`~determined.pytorch.dsat.get_ds_config_from_hparams` helper
+function, which retrieves the configuration from the hyperparameters. You can then pass this
+configuration to deepspeed.initialize as usual:
+
+.. code:: python
+
+   from determined.pytorch.deepspeed import DeepSpeedTrial, DeepSpeedTrialContext
+   from determined.pytorch import dsat
+
+
+   class MyDeepSpeedTrial(DeepSpeedTrial):
+     def __init__(self, context: DeepSpeedTrialContext) -> None:
+         self.hparams = self.context.get_hparams()
+         config = dsat.get_ds_config_from_hparams(self.hparams)
+         model = ...
+         model_parameters= ...
+
+         model_engine, optimizer, train_loader, lr_scheduler = deepspeed.initialize(
+             model=model, model_parameters=model_parameters, config=config
+         )
+
+Using Determined's DeepSpeed Autotune with a :class:`~determined.pytorch.deepspeed.DeepSpeedTrial`
+instance requires no further changes to your user code.
+
+Examples which use DeepSpeed Autotune with HuggingFace Trainer can be found in the
+``examples/deepspeed_autotune/torchvision/deepspeed_trial`` `subdirectory within the Determined
+GitHub Repo
+<https://github.com/determined-ai/determined/tree/master/examples/deepspeed_autotune/torchvision/deepspeed_trial>`__
+.
+
+Core API
+========
+
+When using DeepSpeed Autotune with a Core API experiment, there is one additional change to be made
+following the steps in the :ref:`using_deepspeed_trial` section above.
+
+The ``forward``, ``backward``, and ``step`` methods of the ``DeepSpeedEngine`` class need to be
+wrapped in the :func:`~determined.pytorch.dsat.dsat_reporting_context` context manager. This
+addition ensures that the autotuning metrics from each trial are captured and reported back to the
+Determined master.
+
+Here is an example sketch of ``dsat`` code with Core API:
+
+.. code:: python
+
+   for op in core_context.searcher.operations():
+      for (inputs, labels) in trainloader:
+          with dsat.dsat_reporting_context(core_context, op): # <-- The new code
```

Comment:
It may be worth noting somewhere that when running this outside of a dsat context, that this reporting context function is a no-op